dev/WebView2/WebView2.cpp

```diff
@@ -0,0 +1,1942 @@
+                winrt::CoreWebView2MouseEventVirtualKeys{ static_cast<winrt::CoreWebView2MouseEventVirtualKeys>(GET_KEYSTATE_WPARAM(w_param)) },
+                mouse_data,
+                coords);
+        }
+    }
+    else if ((deviceType == winrt::Windows::Devices::Input::PointerDeviceType::Touch) ||
+             (deviceType == winrt::Windows::Devices::Input::PointerDeviceType::Pen))
+    {
+        const winrt::PointerPoint inputPt{ args.GetCurrentPoint(*this) };
+        winrt::CoreWebView2PointerInfo outputPt = m_coreWebViewEnvironment.CreateCoreWebView2PointerInfo();
+
+        //PEN INPUT
+        if (deviceType == winrt::PointerDeviceType::Pen)
+        {
+            FillPointerPenInfo(inputPt, outputPt);
+        }
+
+        //TOUCH INPUT
+        if (deviceType == winrt::PointerDeviceType::Touch)
+        {
+            FillPointerTouchInfo(inputPt, outputPt);
+        }
+
+        //GENERAL POINTER INPUT
+        FillPointerInfo(inputPt, outputPt, args);
+
+        m_coreWebViewCompositionController.SendPointerInput(winrt::CoreWebView2PointerEventKind{ static_cast<winrt::CoreWebView2PointerEventKind>(message) }, outputPt);
+    }
+}
+
+// The transform is not available in matrix form outside core windows so needed
+// information about the transformation needs to be reconstructed by applying
+// the transform directly to a known set of points.
+// It is assumed that no shear transform is applied and currently rotation is not supported.
+winrt::float4x4 WebView2::GetMatrixFromTransform() {
+    // Calculate transformation assuming 2D only.
+    // Calculate transformed values
+    auto generalTransform = TransformToVisual(nullptr);
+    winrt::Point initialOrigin = winrt::Point(0, 0);
+    winrt::Point translatedOrigin = generalTransform.TransformPoint(initialOrigin);
+
+    winrt::float4x4 outputMatrix{};
+
+    // Assign rotation
+    outputMatrix.m12 = 0.0f;
+    outputMatrix.m13 = 0.0f;
+    outputMatrix.m21 = 0.0f;
+    outputMatrix.m23 = 0.0f;
+    outputMatrix.m31 = 0.0f;
+    outputMatrix.m32 = 0.0f;
+
+    // Assign offsets/translation
+    // This should be the global physical pixel offset to the top left corner of the XAML HWND.
+    outputMatrix.m41 = (translatedOrigin.X * m_rasterizationScale); // X offset
+    outputMatrix.m42 = (translatedOrigin.Y * m_rasterizationScale); // Y offset
+    outputMatrix.m43 = 0.0f; // Z offset
+
+    // Assign scale values
+    // These values will just be 1.0 because Anaheim is getting their values in physical pixels,
+    // so they don't need to do any extra unscaling.
+    outputMatrix.m11 = 1.0f; // X Scale
+    outputMatrix.m22 = 1.0f; // Y Scale
+    outputMatrix.m33 = 1.0f; // Z scale
+
+    // Set to 0 (3D coordinate transform values)
+    outputMatrix.m14 = 0.0f;
+    outputMatrix.m24 = 0.0f;
+    outputMatrix.m34 = 0.0f;
+    // Set to 1 to maintain non-zero det.
+    outputMatrix.m44 = 1.0f;
+
+    return outputMatrix;
+}
```

Comment:
This is no longer used (was needed for older input forwarding API). Please remove.

Reply:
How about you remove in WinUI3 first and then i can copy those deltas over here?